dev/docker/README.md

```diff
@@ -0,0 +1,86 @@
+Apache REEF (incubator) Docker Tests
+====================================
+
+[Apache REEF (incubator)](http://reef.incubator.apache.org/) is 
+an approach to simplify and unify the lower layers of big data systems
+on modern resource managers. This project maintains docker images
+to simulate the target underlying systems as similar as possible.
+
+Docker-based Test Cluster
+-------------------------
+
+| OS           | Java            | Hadoop Version            | Description   | alias     |
```

Comment:
Removing `Oracle` sounds good. I'll remove that!

Microsoft HDInsight 3.1 and 3.2 uses HDP (Hortonworks Data Platform) 2.1.15 and HDP 2.2.7 respectively. As you know, HDP is just one of Apache Hadoop Distribution, but it's important since their Hadoop versions are Hadoop 2.4.0 and Hadoop 2.6.0.

Description about HDInsight is important because the purpose of this PR is providing HDI-similar testing environment for developers and committers. Since Apache REEF has been used in commercial service like HDInsight, testing on HDInsight were mandatory to be committed in some recent PRs.